ccm_web/server/src/api/api.controller.ts

```diff
@@ -95,12 +97,13 @@ export class APIController {
   @ApiQuery({ name: 'course_name', required: false, type: String })
   @Get('admin/sections')
   async getCourseSectionsInTermAsAdmin (
-    @UserDec() user: User,
-      @Query('term_id', ParseIntPipe) termId: number,
-      @Query('instructor') instructor?: string,
-      @Query('course_name') courseName?: string
+    @Query() query: GetSectionsAdminQueryDto,
+      @UserDec() user: User
   ): Promise<CourseWithSections[]> {
-    const result = await this.apiService.getCourseSectionsInTermAsAdmin(user, termId, instructor, courseName)
+    if (query.instructor === undefined && query.course_name === undefined) {
+      throw new BadRequestException('You must specify either instructor or course_name as a URL parameter.')
```

Comment:
The Response in the FE is this Good! I would word this as `instructor_name` instead of `instructor`. That would better for understanding

{
  "statusCode": 400,
  "message": "You must specify either instructor or course_name as a URL parameter.",
  "error": "Bad Request"
}

Reply:
Are you saying he should rename the parameter, because the parameter name and the message are currently consistent.

Reply:
Yes! It will be clear for user if error case hit. I have some similar comments below

Reply:
I can change the parameter name.

As a side note, we will probably want to tell the user that `instructor_name` has to be an exact match, while `course_name` can be a partial match.

Reply:
>  want to tell the user that instructor_name has to be an exact match, while course_name can be a partial match.

Good Idea

Reply:
@pushyamig, here you go: https://github.com/tl-its-umich-edu/canvas-course-manager-next/pull/205/commits/25b6910ca5f91b01712587920e63c85eff307d7c

@chrisrrowland, sorry about the breaking change, but it is minor! It is slightly more clear, since `instructor` could be interpreted as a number ID or a name string. I guess `instructor_login_id` or would be the most clear...

Reply:
Does it really have to  be an exact match or does it support wild carding?  If it does support wildcarding do we want to automatically pad the search text on the backednd like "rowc" => "%rowc%" or whatever the appropriate method would be?  I did try searching for "row%" and "row*" through the front end and it didn't work, but I would assume the canvas documentation would say what was supported if anything is..

Reply:
The instructor value gets passed to a `by_teachers` parameter, not a search parameter, so I believe it has to be exact.
https://canvas.instructure.com/doc/api/accounts.html#method.accounts.courses_api

Reply:
Ok, looks like you'd have to do multiple api calls to lookup matching users or something, if  that's even possible, to allow that.  Thanks.